src/hostapi/jack/pa_jack.c

```diff
@@ -451,6 +452,50 @@ BlockingWaitEmpty( PaStream *s )
 
 /* ---- jack driver ---- */
 
+void replace_string(char *string, size_t buffer_length, char *to_find, char *replacement)
+{
+    char *found_position = strstr(string, to_find);
+    while( found_position ) {
+        size_t original_length = strlen(string);
+        size_t found_length = strlen(to_find);
+        size_t replacement_length = strlen(replacement);
+
+        size_t destination_length = original_length - found_length + replacement_length;
+        if( destination_length >= buffer_length ) {
+            PaUtil_DebugPrint( "Cannot replace %s with %s in %s, buffer size is too small\n",
+                               to_find, replacement, string );
+            return;
+        }
+
+        size_t before_find_length = found_position - string;
+        memmove(string + before_find_length + replacement_length,
+                string + before_find_length + found_length,
+                original_length - before_find_length);
```

Comment:
I think the copied num is wrong. We can also move the null terminator here. How about this. 
memmove(found_position + replacement_length,
                found_position + found_length,
                string + original_length - (found_position + found_length) + 1);

Reply:
yes that works to copy the `\0` terminator